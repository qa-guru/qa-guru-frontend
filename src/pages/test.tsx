import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
