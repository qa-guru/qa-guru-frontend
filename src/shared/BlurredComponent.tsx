import React, { ReactNode } from "react";
import { Box } from "@mui/material";

interface IBlurredComponent {
  children: ReactNode;
}

const style = {
  wrapper: { filter: "blur(3.5px)", pointerEvents: "none" },
};

const BlurredComponent: React.FC<IBlurredComponent> = ({ children }) => {
  return <Box sx={style.wrapper}>{children}</Box>;
};

export default BlurredComponent;
