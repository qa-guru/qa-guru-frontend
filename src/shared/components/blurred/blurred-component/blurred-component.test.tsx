import { MemoryRouter } from "react-router-dom";
import { render } from "test/utils-test";

import BlurredComponent from "./blurred-component";

describe("BlurredComponent", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <BlurredComponent>
          <div>Sample content</div>
        </BlurredComponent>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
