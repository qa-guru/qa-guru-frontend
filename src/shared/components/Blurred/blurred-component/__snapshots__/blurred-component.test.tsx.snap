// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlurredComponent > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1nsqiu3"
  >
    <div>
      Sample content
    </div>
  </div>
</DocumentFragment>
`;
